container.go

```diff
@@ -50,6 +53,20 @@ type Container struct {
 	graph.Graph
 }
 
+// InvokeOnce only allows function invocation once to register the
+// return types. If return types are already registered, specific error
+// is returned to the caller
+func (c *Container) InvokeOnce(t interface{}) error {
+	ctype := reflect.TypeOf(t)
+	if ctype.Kind() != reflect.Func {
+		return errors.Wrapf(errParamType, _invokeErr, ctype)
+	}
+	if err := c.ValidateReturnTypes(ctype, true); err != nil {
+		return ErrInvokeOnce
+	}
+	return c.Invoke(t)
```

Comment:
This is what I want to know:

```
type type1 struct {}
type type2 struct{}
type type3 struct{}
func newT1(type2, type3) *type1 {}
func newT2() *type2 {}
func newT3() *type3{}

c := dig.New()
c.InvokeOnce(newT1) // calls newT2 and newT3 first to satisfy deps, then newT1
c.InvokeOnce(newT2) // <--- here, will actually execute T2?
```

And I'd actually really like a UT for that. To make sure all constructors are called only once.